### PR TITLE
fixLibtool(): replace /usr/bin/file in ./configure, add file to common-path.nix

### PR DESCRIPTION
**Pull request by ghost**
This is a much cleaner replacement for #166879, but it will cause everything to rebuild.

###### Description of changes

libtool's `libtool.m4` script assumes that `file` is available, and can be found at `/usr/bin/file` (this path is [hardwired](https://git.savannah.gnu.org/cgit/libtool.git/tree/m4/libtool.m4?h=v2.4.6&id=f09d00cbcf924c378573163e244fffeb8d28005f#n1346)).  Furthermore, this script is vendored into the `./configure` scripts of an enormous number of packages.  Without this PR, you will frequently see errors like this during the `configurePhase` when the sandbox is enabled:

```
  ./configure: line 9595: /usr/bin/file: command not found
```

Due mostly to luck, this error does not affect native compiles on nixpkgs' two most popular platforms, `x86_64-linux` and `aarch64-linux`. However it will cause incorrect linker flag detection and a failure to generate shared libraries for sandboxed cross-builds to a `x86_64-linux` host as well as any sandboxed build (cross or native) for the following hosts: `x86_64-freebsd`, `*-hpux`, `*-irix`, `mips64*-linux`, `powerpc*-linux`, `s390x-linux`, `s390x-tpf`, `sparc-linux`, and `*-solaris`.

This PR fixes the problem by adding [an extra line](https://github.com/NixOS/nixpkgs/pull/168413/commits/0bec8faa22ddeb050b5f769b224a81d8cdd9c7d4#diff-7c7ca80928136cfc73a02d5b28350bd900e331d6d304857053ffc9f7beaad576R1004) to the `if`-block which calls `fixLibtool()` in `pkgs/stdenv/generic/setup.sh`.  This extra line will scan the unpacked source code for executable files named `configure` which contain the following text:

```
GNU Libtool is free software; you can redistribute it and/or modify
```

This text is taken to be an indicator of a vendored `libtool.m4`.  When it is found, the configure script containing it is subjected to `sed -i s_/usr/bin/file_file_` which replaces all occurrences of `/usr/bin/file` with `file`, and the mtime of the file is preserved across this replacement.

Additionally, the `file` package is now considered to be part of `stdenv`.  It has been added to `common-path.nix` so that the `file` binary will be found in the `$PATH` of every build, except for the bootstrap-tools and the first few stages of stdenv boostrapping.

Verified no regressions under:

```
  nix-build --arg pkgs 'import ./. {}' ./lib/tests/release.nix
```

This commit allows the following commands to complete, which should enable Hydra to produce bootstrap-files for mips64el:

```
  nix-build \
    --option sandbox true \
    --option sandbox-fallback false \
    pkgs/top-level/release-cross.nix \
    -A bootstrapTools.mips64el-linux-gnuabi64.build

  nix-build \
    --option sandbox true \
    --option sandbox-fallback false \
    . \
    -A pkgsCross.mips64el-linux-gnuabi64.nix_2_4
```

CC: @sternenseemann, @SuperSandro2000

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
  - [x] powerpc64le-linux
  - [x] mips64el-linux
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
```
[nix-shell:~/.cache/nixpkgs-review/rev-5952698e53b489a2e92e5b87b765a1577678ed90]$ cat report.json
{
    "blacklisted": [],
    "broken": [],
    "built": [],
    "failed": [],
    "non-existant": [],
    "pr": null,
    "system": "x86_64-linux",
    "tests": []
}
[nix-shell:~/.cache/nixpkgs-review/rev-5952698e53b489a2e92e5b87b765a1577678ed90]$ cat report.md
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
```
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by a-m-joseph** [comment]
I've verified no regressions on x86_64-linux.  I *think* I fixed the darwin build failure that ofborg caught last night; we'll find out shortly.

**Comment by a-m-joseph** [comment]
I'm trying to reproduce ofborg's failure.

**Comment by a-m-joseph** [comment]
@ofborg eval libtool

**Comment by a-m-joseph** [comment]
@ofborg build libtool

**Comment by a-m-joseph** [comment]
Ofborg said:

```
help2man: can't get `--help' info from automake-1.16
Try `--no-discard-stderr' if option outputs to stderr
make: *** [Makefile:3743: doc/automake-1.16.1] Error 127
builder for '/nix/store/ny9apfxxwfz9lkzca961dlsq1mdn2bw8-automake-1.16.5.drv' failed with exit code 2
```

and yet on my own machine,

```
$ nix-build /nix/store/ny9apfxxwfz9lkzca961dlsq1mdn2bw8-automake-1.16.5.drv
/nix/store/8q6zv25b9ff461kd1yib0gqxsc44x2xs-automake-1.16.5
```

Huh.

**Comment by a-m-joseph** [comment]
@ofborg build automake

**Comment by a-m-joseph** [comment]
> @ofborg build automake

[Succeeded](https://logs.nix.ci/?attempt_id=77fec057-c877-41be-9441-2a7317361d30&key=nixos%2Fnixpkgs.168413).

WTF?

**Comment by a-m-joseph** [comment]
@eval

**Comment by a-m-joseph** [comment]
@ofborg eval

File: pkgs/development/libraries/libidn2/default.nix
Comment (Mindavi) [inline pkgs/development/libraries/libidn2/default.nix:35]: Because GTKDOCIZE is now always overridden, are these tools still needed for darwin?

File: pkgs/stdenv/linux/default.nix
Comment (Mindavi) [inline pkgs/stdenv/linux/default.nix:412]: Did you check that file doesn't reference any bootstrap files?

**Review by Mindavi — APPROVED** [review]
Got 2 small questions/remarks, but in general this looks very good and is also well documented. I really like to see people working on the cross support and this looks like a good step forward for a lot of platforms.

File: pkgs/stdenv/linux/default.nix
Comment (a-m-joseph) [reply pkgs/stdenv/linux/default.nix:412]: Ah, no I did not.  Thank you for reminding me!

```
$ git log --oneline | head -n 2
dbdca8de867 libidn2: facilitate its self-autoreconf-ing behavior
0bec8faa22d fixLibtool(): patch ./configure, add `file` to common-path.nix
$ nix-store -q --requisites $(nix-build $(nix-instantiate . -A stdenv)) | grep bootstrap-tools
```

So, it looks like we're good.

Is there a reason why `pkgs/stdenv/linux/default.nix` doesn't use `disallowedRequisites` to make this check automatic?  For example, like #168590.  I suppose @Ericson2314 could answer this.

File: pkgs/development/libraries/libidn2/default.nix
Comment (a-m-joseph) [reply pkgs/development/libraries/libidn2/default.nix:35]: Good question; I don't really know.  I don't have Darwin hardware to test so I wanted to err on the side of not breaking anything.

Comment (Mindavi) [reply pkgs/development/libraries/libidn2/default.nix:35]: That's fair.

**Comment by a-m-joseph** [comment]
> @ofborg eval

So apparently the earlier ofborg failure was some kind of glitch in the matrix.  Nothing was pushed to this PR between the original ofborg failure and the eval above, which succeeded on all checks.

**Comment by a-m-joseph** [comment]
It looks like this should wait until after staging closes tomorrow and reopens on 2022-May-15.  Comments still welcome in the meantime.

**Comment by a-m-joseph** [comment]
Hello, now that staging is open again for post-22.05 merges I have undrafted this PR.

Committers, please consider merging this soon.  It is on the critical path blocking Hydra from being able to produce a mips64le `bootstrap-files` whose hash we can reference.

I have been using this commit on all of my machines (x86_64, aarch64, mips64el, powerpc64le) for the past month on top of `nixos-unstable` without any problems.

File: pkgs/development/libraries/libidn2/default.nix
Comment (sternenseemann) [inline pkgs/development/libraries/libidn2/default.nix:32]: Where exactly is that and how does autoreconfHook solve this? I'm a bit confused by this and I wasn't able to find what you are referring to in the libidn2 tarball.

**Review by sternenseemann — APPROVED** [review]
Built

* sway
* systemd
* pandoc
* emacs

On `x86_64-linux` and all seems well. Will try some cross builds now to make sure that that did not regress.

**Comment by a-m-joseph** [comment]
@SuperSandro2000 thanks for looking at this!  I have resolved the merge conflict.

File: pkgs/development/libraries/libidn2/default.nix
Comment (a-m-joseph) [reply pkgs/development/libraries/libidn2/default.nix:35]: Here is the error message you get without `autoreconfHook`; I'm still working on pinpointing the conditional that `libidn2` uses to decide that this is necessary:

```
make[3]: Entering directory '/build/libidn2-2.3.2/doc'
 cd .. && /nix/store/p4s4jf7aq6v6z9iazll1aiqwb34aqxq9-bootstrap-tools/bin/bash /build/libidn2-2.3.2/build-aux/missing automake-1.16 --gnu doc/Makefile
/build/libidn2-2.3.2/build-aux/missing: line 81: automake-1.16: command not found
WARNING: 'automake-1.16' is missing on your system.
         You should only need it if you modified 'Makefile.am' or
         'configure.ac' or m4 files included by 'configure.ac'.
         The 'automake' program is part of the GNU Automake package:
         <https://www.gnu.org/software/automake>
         It also requires GNU Autoconf, GNU m4 and Perl in order to run:
         <https://www.gnu.org/software/autoconf>
         <https://www.gnu.org/software/m4/>
         <https://www.perl.org/>
make[3]: *** [Makefile:1414: Makefile.in] Error 127
make[3]: Leaving directory '/build/libidn2-2.3.2/doc'
make[2]: *** [Makefile:2122: Makefile.gdoc] Error 2
make[2]: Leaving directory '/build/libidn2-2.3.2/doc'
make[1]: *** [Makefile:1401: all-recursive] Error 1
make[1]: Leaving directory '/build/libidn2-2.3.2'
make: *** [Makefile:1309: all] Error 2
note: keeping build directory '/tmp/nix-build-libidn2-2.3.2.drv-1'
```

Comment (a-m-joseph) [reply pkgs/development/libraries/libidn2/default.nix:35]: It *is* the file mtimes.

This has led me to discover an even cleaner way to do this, which will be robust in the face of any other packages which do what `libidn2` does (although as I mentioned I have not found any).

Since I will be adding another commit to this PR, I'm going to "resolve" this conversation and continue below at the bottom of the discussion-chain so the other reviewers see the changes.

Thank you for prompting me to dig further into this!

**Comment by a-m-joseph** [comment]
It turns out that `libidn2` decides to "autoreconf itself" based on file modification times.

Therefore, simply ensuring that we do not change any file mtimes in `fixLibtool()` makes the changes to libidn2 unnecessary.  This is also a more general solution; if any other package does what `libidn2` does, then preserving mtimes will work there too without a package-specific fix.

I also realized that `fixLibtool()` is called more than once -- it is invoked for every `ltmain.sh` file in the source tarball.  Therefore we should do the `configure`-patching before/after the loop which invokes `fixLibtool()`, rather than inside `fixLibtool()`.

In the push following this commit, I will:

1. Drop the commit which affected `libidn2/default.nix`.
2. Add a commit which
   a. Preserves the mtimes of files upon which we invoke `sed -i`
   b. Moves the `find ... -execdir sed -i` out of `fixLibtool()` and into the `if !dontFixLibtool { ... }` block which contains the loop which calls `fixLibtool()`.

If you have already reviewed this PR, you only need to re-review the two new commits.

@Mindavi and @sternenseemann had already approved this PR prior to this change.  I have re-requested reviews from them; once they acknowledge/reapprove I will squash the PR down to a single commit.

Thanks again to @sternenseemann for [prompting me in his review](https://github.com/sternenseemann) to dig further into the libidn2 situation, which resulted in this PR becoming much cleaner and more robust.

File: pkgs/stdenv/generic/setup.sh
Comment (jtojnar) [inline pkgs/stdenv/generic/setup.sh:1028]: The latest version of libtool already has this fixed by our very own @aakropotkin, and I would say that when a package has pre-generated build scripts using ancient version of `libtool`, it likely uses other old macros as well so this might not be the only issue. Did you consider just forcing building for all packages, e.g. by running `make distclean`?

Though, I recognize that this might be hard due to bootstrapping, removing files that we are not able to generate from the tarball, different projects having [different `autogen.sh` styles](https://www.sourceware.org/autobook/autobook/autobook_43.html), not supporting autoreconf, so we might still want to go with this for the time being.

Comment (jtojnar) [reply pkgs/stdenv/generic/setup.sh:1028]: Actually, just remembered Debian has that as policy so we could look there: https://wiki.debian.org/Autoreconf

Comment (a-m-joseph) [reply pkgs/stdenv/generic/setup.sh:1028]: > The latest version of libtool already has this fixed by our very own @aakropotkin

Yes I know, I wrote [the PR](https://github.com/NixOS/nixpkgs/pull/167071) that bumped the version in nixpkgs!

It is wonderful that libtool 2.4.7 no longer forces `/usr/bin/file` to be hardcoded.  Unfortunately it does not fix the entire problem.  Here are the two remaining failure modes:

1. If upstream authors set `FILECMD` when they run the autotools, whatever they set it to will get hardcoded into the vendored source tarball.
2. If the source tarball was produced with libtool<=2.4.6.  This is 100% of all upstream software released before 2022-Mar-17.

> ancient version of `libtool`

I guess if "ancient" means "more than [two months old](https://savannah.gnu.org/forum/forum.php?forum_id=10139)"...

> Did you consider just forcing building for all packages, e.g. by running make distclean?

If you mean adding `autoreconfHook` then yes, that was one of the first things I tried.  Unfortunately if you blindly `autoreconfHook` every package you get tons of failures because they only work when `autoreconf`ed with exactly the version that the original author used.  Here is [a great example of the breakage](https://github.com/NixOS/nixpkgs/pull/163401#issuecomment-1072040524).

**Comment by Mindavi** [comment]
I'm okay with the workaround for now, autoreconf-ing is probably a pain to do for all packages.

In general it's good to have this to improve exotic platform support and to have better cross-compilation support from more rare platforms.

File: pkgs/stdenv/generic/setup.sh
Comment (sternenseemann) [inline pkgs/stdenv/generic/setup.sh:1029]: We handle quoting in a very lax way in nixpkgs, but I think at least in places like `setup.sh` we should do it properly.

**Review by sternenseemann — APPROVED** [review]
To what extent has the darwin stdenv been tested?

**Review by Mindavi — APPROVED** [review]

File: pkgs/stdenv/generic/setup.sh
Comment (a-m-joseph) [reply pkgs/stdenv/generic/setup.sh:1029]: You mean if $TMPDIR contains spaces, I presume?  I guess that is possible.

I am generally pretty paranoid about shell quoting issues which is why I wrote the `find` invocation the way I did.

Comment (a-m-joseph) [reply pkgs/stdenv/generic/setup.sh:1029]: https://github.com/NixOS/nixpkgs/pull/168413/commits/43a087cacce9b4fd9e4bd965441320f06b9b35a6

**Comment by a-m-joseph** [comment]
> added the ofborg-internal-error label 3 hours ago

Apparently resistance is *not* futile!

**Comment by a-m-joseph** [comment]
> > To what extent has the darwin stdenv been tested?
> 
> I lack darwin hardware, so this is the best I can do right now:
> 
> @ofborg build libidn2

> libidn2 on aarch64-darwin — [Success](https://github.com/NixOS/nixpkgs/pull/168413/checks?check_run_id=6619066807)

**Comment by cole-h** [comment]
@ofborg eval

File: pkgs/stdenv/common-path.nix
Comment (Artturin) [inline pkgs/stdenv/common-path.nix:22]: a comment like 
https://github.com/NixOS/nixpkgs/blob/89a3d054790bbcf6c9d73394cd1c541466c5a7c8/pkgs/tools/compression/gzip/default.nix#L7-L10
should be added to file.

Comment (Artturin) [reply pkgs/stdenv/common-path.nix:22]: @matthewbauer can you confirm whether or not that is necessary? https://github.com/NixOS/nixpkgs/pull/91636

**Comment by a-m-joseph** [comment]
> Mindavi merged commit f174277 into NixOS:staging yesterday

Thank you thank you thank you!!!!

We should have a bootstrap for mips64el very soon now.  Thank you again!

File: pkgs/stdenv/common-path.nix
Comment (a-m-joseph) [reply pkgs/stdenv/common-path.nix:22]: > a comment like
> 
> https://github.com/NixOS/nixpkgs/blob/89a3d054790bbcf6c9d73394cd1c541466c5a7c8/pkgs/tools/compression/gzip/default.nix#L7-L10
> 
> should be added to file.

https://github.com/NixOS/nixpkgs/pull/175343

**Comment by aakropotkin** [comment]
Shit is it writing `FILECMD=/definitely/evil/bin/file` as a hard coded path
into people's source tarballs?

If it is let me know so I can patch.

Or is the issue that `libtoolize` now requires the runtime `coreutils`
store path to match the one used to build `libtool`?

**Comment by aakropotkin** [comment]
Oh jesus, the bootstrap paradox of `file` needing `file` is fun.

We'll resolve this, but honestly I might rewrite `magiccmd` ( the reason
`file` is needed ) to use `read` to check file magic to avoid this whole
mess:

```
# Test if a file is an AR archive.
 function isAR() {
   local fn="$1"
   local fd
   local magic
   exec {fd}<"$fn"
   # In Zsh `read -k NUM' behaves like Bash's `read -n NUM'.
   if test "${ZSH_VERSION+set}" = set; then
     read -r -k 7 -u "$fd" magic
   else
     read -r -n 7 -u "$fd" magic
   fi
   read -r -n 7 -u "$fd" magic
   exec {fd}<&-
   # First seven characters should be "!<arch>"
   if test "$magic" = $'\041\074arch\076'; then return 0; else return 1; fi
 }
```

**Comment by sternenseemann** [comment]
> If it is let me know so I can patch.

AIUI it is [already patched](https://git.savannah.gnu.org/cgit/libtool.git/commit/m4/libtool.m4?id=da2e352735722917bf0786284411262195a6a3f6), but we need a workaround for tarballs generated with older libtool versions.

**Comment by a-m-joseph** [comment]
@aakropotkin, thank you again so much for fixing the upstream issue in libtool.  Let's be honest, this PR (#168413) is basically a hack.  Thanks to your work on libtool we will be able to revert this hack ~3-4 years from now instead of infinitely-many years from now.

> Shit is it writing `FILECMD=/definitely/evil/bin/file` as a hard coded path into people's source tarballs?

I believe you are referring to my comment upthread about other failure modes, where I wrote:

> If upstream authors set `FILECMD` when they run the autotools, whatever they set it to will get hardcoded into the vendored source tarball.

I swear that I experienced `FILECMD=/foobarbaz libtoolize; aclocal; autoconf` leaking "`/foobarbaz`"es into `./configure`, but I am unable to reproduce this, so I must have been imagining it.

However there are still some hardcoded `/usr/bin/file`s in `libtool` itself:

```
nix@moore:/nix/nixpkgs$ nix-build . -A libtool
/nix/store/cykvcas3pyipzmwvnskc0mb1x43nmdfs-libtool-2.4.7
nix@moore:/nix/nixpkgs$ rg /usr/bin/file /nix/store/cykvcas3pyipzmwvnskc0mb1x43nmdfs-libtool-2.4.7
/nix/store/cykvcas3pyipzmwvnskc0mb1x43nmdfs-libtool-2.4.7/share/libtool/build-aux/ltmain.sh
8543:               if /usr/bin/file -L $add 2> /dev/null |

/nix/store/cykvcas3pyipzmwvnskc0mb1x43nmdfs-libtool-2.4.7/bin/libtool
9064:               if /usr/bin/file -L $add 2> /dev/null |
```

Perhaps those don't matter.

> Oh jesus, the bootstrap paradox of `file` needing `file` is fun.

:)

> We'll resolve this, but honestly I might rewrite `magiccmd` ( the reason `file` is needed ) to use `read` to check file magic to avoid this whole mess:

That would be wonderful.  You'll need a bit more than the four "magic" bytes, but not much.  Reading the first nine bytes will give you the CPU, endianness, and ABI, which should be enough:

https://en.wikipedia.org/wiki/Executable_and_Linkable_Format#File_header

**Comment by mehmooda** [comment]
If file is now in stdenv does this mean that it can be removed from mingw [stdenv/cross](https://github.com/NixOS/nixpkgs/blob/766f5ffb761bc916ea0f270f472d04ab30664d52/pkgs/stdenv/cross/default.nix#L88)?

**Comment by SuperSandro2000** [comment]
Worth a try

**Comment by a-m-joseph** [comment]
> If file is now in stdenv does this mean that it can be removed from mingw [stdenv/cross](https://github.com/NixOS/nixpkgs/blob/766f5ffb761bc916ea0f270f472d04ab30664d52/pkgs/stdenv/cross/default.nix#L88)?

Yes.  Thanks for noticing this.

I will take the blame if removing it breaks the world: #176597  

Side question: are there any tools to make it easier to trace the provenance of a line of code using `git blame` when it has been moved around by refactoring commits?  Sort of like `git rebase --continue` where you're alternating between the tool doing as much as it can before stopping for human assistance.  Just using `git grep` doesn't work too well because the search string might change a few times during the search as refactoring alter the identifier names change (due to refactoring).

**Comment by a-m-joseph** [comment]
> Bisect claims that [97c4382](https://github.com/NixOS/nixpkgs/commit/97c43828fb7e016b4ee8fe434bc4d5e0b8a8b4be) `fixLibtool(): patch ./configure, add `file` to common-path.nix` broke `abuild` in `staging` and `staging-next`:

I am investigating.

Don't hesitate to revert this if it's blocking other people's progress.

You can expect a more detailed response from me within the next 12 hours (likely much sooner than that).

**Comment by a-m-joseph** [comment]
> Bisect claims that [97c4382](https://github.com/NixOS/nixpkgs/commit/97c43828fb7e016b4ee8fe434bc4d5e0b8a8b4be) `fixLibtool(): patch ./configure, add `file` to common-path.nix` broke `abuild` in `staging` and `staging-next`:

Fixed: https://github.com/NixOS/nixpkgs/pull/177682

**Comment by trofi** [comment]
Bisect claims `epson-escpr2` was broken by 97c43828fb7e016b4ee8fe434bc4d5e0b8a8b4be `fixLibtool(): patch ./configure, add file to common-path.nix`:

```
$ nix build -f. -L epson-escpr2
...
epson-inkjet-printer-escpr2> patching sources
epson-inkjet-printer-escpr2> applying patch /nix/store/ljqnzvnhv6v0c4pvwry3617fp8j7q3bv-cups-filter-ppd-dirs.patch
epson-inkjet-printer-escpr2> patching file configure
epson-inkjet-printer-escpr2> configuring
epson-inkjet-printer-escpr2> fixing libtool script ./ltmain.sh
epson-inkjet-printer-escpr2> mktemp: (null): Invalid argument
```

**Comment by a-m-joseph** [comment]
> Bisect claims `epson-escpr2` was broken by [97c4382](https://github.com/NixOS/nixpkgs/commit/97c43828fb7e016b4ee8fe434bc4d5e0b8a8b4be) `fixLibtool(): patch ./configure, add file to common-path.nix`:

Ugh, this is another case of a package fooling `setup.sh` into using the `busybox` version of a command.  In this case, `mktemp`:

```
$ busybox mktemp --help
BusyBox v1.35.0 () multi-call binary.

Usage: mktemp [-dt] [-p DIR] [TEMPLATE]

Create a temporary file with name based on TEMPLATE and print its name.
TEMPLATE must end with XXXXXX (e.g. [/dir/]nameXXXXXX).
```

Notice the special requirement on the number of `X`es.

**Comment by a-m-joseph** [comment]
> Bisect claims `epson-escpr2` was broken by [97c4382](https://github.com/NixOS/nixpkgs/commit/97c43828fb7e016b4ee8fe434bc4d5e0b8a8b4be) `fixLibtool(): patch ./configure, add file to common-path.nix`:

Fix: https://github.com/NixOS/nixpkgs/pull/178626

**Comment by aakropotkin** [comment]
This thread has grown quite a bit, and I'll admit I've fallen behind on reading it.

If there's any issues that could or should be resolved upstream in `libtool` please tag me so I will see them.

I'm sure there's a mix of "old builds broke because of issues in their usage" vs "the `libtool` maintainer used a GNU specific flag in a script and blew up <niche distro>". I did what I could in terms of testing before the release, but I'll be transparent and admit that I was only able to run NixOS, Debian, Ubuntu, OSX ( aarch64 and x86_64 ), Windows ( pretty unsuccessfully ) and RHEL. And for all of those I just ran with the default tools on a box. Other than that I relied on `shellcheck` and the GNU shell portability manual, but I would be unsurprised to hear that I used a flag/utility that isn't available for certain systems.

**Comment by a-m-joseph** [comment]
> This thread has grown quite a bit, and I'll admit I've fallen behind on reading it.

> If there's any issues that could or should be resolved upstream in `libtool` please tag me so I will see them.

You can unsubscribe if you want.  I will ping @aakropotkin if anything comes up that needs upstream action, but I doubt that it will.

Thank you again very much for adding `FILECMD` in libtool 2.4.7!

**Comment by aakropotkin** [comment]
I wrote it so that you can either autoreconf or `export FILECMD="$( command
command -v file:)"` IIRC. Let me know if that's not the case.

**Pull request merged**
